### PR TITLE
Add notebook documenting usage of core layer

### PR DESCRIPTION
**Pull request by vinisalazar**
Hi,

This PR adds `notebooks/04-refactor-update.ipynb` documenting how to use public methods from the core layer.

I intend to improve this in the future, but I just added a basic example for now.

This PR contains 3ed7d61057fa7231ac8c96b39de6f9bcae3dd667 from #281.

Thanks

### CODE REVIEW

File: notebooks/04-refactor-update.ipynb [hunk 1]
```diff
@@ -0,0 +1,135 @@
+{
```
Comment (ocefpaf) [inline notebooks/04-refactor-update.ipynb:1]: <p>Can you remove this empty cell? We may want to use this in our docs.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ioos/erddapy/pull/282/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

**Review by ocefpaf — APPROVED** [review]
I'll wait a bit to see if Alex want to comment but I believe they are getting ready for Thanksgiving. I'll merge this Friday if we don't hear from him.

**Comment by abkfenris** [comment]
It looks like it is still partly referring to the top level import when that has/will change.

**Comment by vinisalazar** [comment]
> It looks like it is still partly referring to the top level import when that has/will change.

Sorry, I didn't understand, could you please clarify?

**Comment by vinisalazar** [comment]
Never mind, fixed it

**Pull request merged**